asamisellah/Maintenance-Tracker
# #158238202 Create tests for Admin Endpoints

### What does this PR do?
Create tests for admin endpoints fthat approve, disapprove or resolve a request made by a user

### Description of Task to be completed?
Test the following endpoints

METHOD | ENDPOINT | FUNCTIONALITY 
------------ | -------------|--------------
PUT | /api/v1/requests/</requestId>/approve | Updates status of request to approved if pending
PUT | /api/v1/requests/</requestId>/disapprove | Updates status of request to disapproved if pending
PUT | /api/v1/requests/</requestId>/resolve | Updates status of request to resolved if approved

Modify tests for already existing endpoints

### How should this be manually tested?
1. clone the [repo](https://github.com/asamisellah/Maintenance-Tracker)
2. open the folder in your text editor or IDE
3. `git checkout -b new-branch`
4. `git fetch ch-env-setup-157930685`

*Setup*
Make sure virtual environment is installed. Open terminal and run the following commands
1. `cd maintenance-tracker`
2.  `virtualenv venv` to create virtual environment
3.  `source venv/bin/activate` to run environment
4. `pip install -r requirements.txt`

*Database*
Prerequisites: Should have Postgres installed
Open terminal and run
``` $ sudo -u postgres psql
    $ CREATE DATABASE mtracker_test;
    $ \q   
    $ python t_schema.py
    $ pytest
```

### Any background context you want to provide?
None

### What are the relevant pivotal tracker stories
[#158238202](https://www.pivotaltracker.com/story/show/158238202/comments/190533443)
